src/OptionsResolver.php

```diff
@@ -0,0 +1,401 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver as BaseOptionsResolver;
+
+/**
+ * @internal
+ */
+final class OptionsResolver extends BaseOptionsResolver
```

Comment:
Sadly, I do not like the approach here. Don't get me wrong, I don't have better solution ( at least yet ), but copy-paste all private properties and duplicate the code to handle them is a pretty big boilerplate. Imagine that sb would fix a bug in original implementation - we would not be aware of this in our duplicated code.

`OptionsResolver` class is not final, but yet it uses private properties.
Could you propose to make the properties protected instead ?